src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
+  const newDateSeparator = toFormat[3];
+
+  const thresholdYear = 30;
+
+  if (oldYear.format !== newYear.format) {
+    if (oldYear.format.length < newYear.format.length) {
+      year =
+        dateParts[oldYear.place] < thresholdYear
+          ? `20${dateParts[oldYear.place]}`
+          : `19${dateParts[oldYear.place]}`;
+    } else {
+      year = dateParts[oldYear.place].slice(2);
+    }
+  } else {
+    year = dateParts[oldYear.place];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        formatedDate[i] = year;
+        break;
+      case 'MM':
+        formatedDate[i] = dateParts[fromFormat.indexOf('MM')];
+        break;
+      case 'DD':
+        formatedDate[i] = dateParts[fromFormat.indexOf('DD')];
+        break;
+    }
+  }
+
+  return formatedDate.join(newDateSeparator);
+}
+
+function yearFormatAndPlace(pattern) {
+  const longYearFormat = 'YYYY';
+  const shortYearFormat = 'YY';
+  let format = '';
+  let place = -1;
+
+  for (let i = 0; i < pattern.length; i++) {
+    if (pattern[i] === longYearFormat || pattern[i] === shortYearFormat) {
+      format = pattern[i];
+      place = i;
+      break;
+    }
+  }
+
+  return {
+    format,
+    place,
+  };
 }
```

Comment:
The function `yearFormatAndPlace` does not validate that the `pattern` array contains the expected elements. It assumes that either 'YYYY' or 'YY' will be present without checking the length or contents of the array, which could lead to incorrect behavior if the input does not meet these expectations.